### PR TITLE
Multi-file-config

### PR DESCRIPTION
**Pull request by alex-schneider**
reads and merges configuration files from a given directory 

---
<details>
    <summary>Reviewer checklist</summary>
    <ul>
        <li>Read PR description: a summary about the changes is required</li>
        <li>Changelog updated</li>
        <li>Documentation: docs/{Reference, Cli, ...}, Docker and cli help/usage</li>
        <li>Pulled branch, manually tested</li>
        <li>Verified requirements are met</li>
        <li>Reviewed the code</li>
        <li>Reviewed the related tests</li>
    </ul>
</details>

### CODE REVIEW

File: config/configload/merge.go [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+
+	return &hclsyntax.Block{
+		Type: "definitions",
+		Body: &hclsyntax.Body{
+			Blocks: blocks,
+		},
+	}, nil
+}
+
+func mergeAttributes(blockName string, bodies []*hclsyntax.Body) (*hclsyntax.Block, error) {
```
Comment (johakoch) [inline config/configload/merge.go:347]: no error is thrown by this function

File: config/configload/merge.go [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+
+	for _, body := range bodies {
+		for _, outerBlock := range body.Blocks {
+			if outerBlock.Type == "definitions" {
+				for _, innerBlock := range outerBlock.Body.Blocks {
+					if definitions[innerBlock.Type] == nil {
+						definitions[innerBlock.Type] = make(data)
+					}
+
+					if len(innerBlock.Labels) == 0 {
```
Comment (johakoch) [inline config/configload/merge.go:298]: You expect exactly one label for merging, right?
So `!= 1` instead of `== 0`?

File: docs/CLI.md [hunk 3]
```diff
@@ -19,6 +19,7 @@ Couper is build as binary called `couper` with the following commands:
 | Argument             | Default      | Environment                | Description                                                                               |
 | :------------------- | :----------- | :------------------------- | :---------------------------------------------------------------------------------------- |
 | `-f`                 | `couper.hcl` | `COUPER_FILE`              | File path to your Couper configuration file.                                              |
+| `-d`                 | -            | `COUPER_FILE_DIRECTORY`    | File path to your Couper configuration files directory.                                   |
```
Comment (johakoch) [inline docs/CLI.md:22]:
```suggestion
| `-d`                 | -            | `COUPER_FILE_DIRECTORY`    | Path to your Couper configuration files directory.                                   |
```

File: docs/CLI.md [hunk 4]
```diff
@@ -28,6 +29,20 @@ Couper is build as binary called `couper` with the following commands:
 
 _Note_: `log-format`, `log-level` and `log-pretty` also map to [settings](REFERENCE.md#settings-block).
 
+_Note_: Couper can be started with both, `-f` and `-d` arguments. The path of `-f <file>`
+is the working directory of Couper. If `-d <dir>` argument is given without the `-f <file>`,
```
Comment (johakoch) [inline docs/CLI.md:33]:
```suggestion
_Note_: Couper can be started with both, `-f` and `-d` arguments. The path of `-f <file>`
determines the working directory of Couper. If `-d <dir>` argument is given without the `-f <file>`,
```

File: config/configload/merge.go (hunk 2 above)
Comment (johakoch) [reply config/configload/merge.go:298]: But the blocks in `definitions` only have exactly one label.

**Comment by johakoch** [comment]
Needs test using config files in server/testdata/multi/server.

**Comment by johakoch** [comment]
Does the merging take **all** block labels into account?

**Review by malud — APPROVED** [review]

**Pull request merged**